tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/PersonNameFormatter.java

```diff
@@ -981,263 +1041,273 @@ private <T> void addToLabel(T item, StringBuilder sb) {
+//                && formalities.equals(that.formalities)) {
+//                return new ParameterMatcher(orders, Sets.union(lengths, that.lengths), usages, formalities);
+//            } else if (orders.equals(that.orders)
+//                && lengths.equals(that.lengths)
+//                && formalities.equals(that.formalities)) {
+//                return new ParameterMatcher(orders, lengths, Sets.union(usages, that.usages), formalities);
+//            } else if (orders.equals(that.orders)
+//                && lengths.equals(that.lengths)
+//                && usages.equals(that.usages)) {
+//                return new ParameterMatcher(orders, lengths, usages, Sets.union(formalities, that.formalities));
+//            }
+//            return null;
+//        }
+//
+//        public ParameterMatcher slim() {
+//            return new ParameterMatcher(
+//                orders.equals(Length.ALL) ? ImmutableSet.of() : orders,
+//                    lengths.equals(Formality.ALL) ? ImmutableSet.of() : lengths,
+//                        usages.equals(Usage.ALL) ? ImmutableSet.of() : usages,
+//                            formalities.equals(Order.ALL) ? ImmutableSet.of() : formalities
+//                );
+//        }
+//
+//        /**
+//         * Compact a set of FormatParameters into a (smaller) collection of ParameterMatchers
+//         */
+//        public static Set<ParameterMatcher> compact(Collection<FormatParameters> expanded) {
+//            Set<ParameterMatcher> result = new TreeSet<>();
+//            for (FormatParameters item : expanded) {
+//                result.add(new ParameterMatcher(item));
+//            }
+//            // try merging each pair
+//            // if we can merge, then start over from the top
+//            // look at optimizing later
+//            main:
+//                while (true) {
+//                    for (ParameterMatcher item1 : result) {
+//                        for (ParameterMatcher item2: result) {
+//                            if (item1 == item2) { // skip ourselves
+//                                continue;
+//                            }
+//                            ParameterMatcher item12 = item1.merge(item2); // merge if possible
+//                            if (item12 != null) {
+//                                result.remove(item1);
+//                                result.remove(item2);
+//                                result.add(item12);
+//                                continue main; // retry everything
+//                            }
+//                        }
+//                    }
+//                    break;
+//                }
+//
+//            // now replace any "complete" items by empty.
+//            Set<ParameterMatcher> result2 = new TreeSet<>();
+//            for (ParameterMatcher item1 : result) {
+//                result2.add(item1.slim());
+//            }
+//            return ImmutableSet.copyOf(result2);
+//        }
+//
+//    }
+
```

Comment:
Do you need to keep all this commented-out code in here?  If you can just get it out of a previous revision in git, that'd be cleaner.  (Or is this just still here until you're completely done with this simplification effort, and THEN you'll delete it?)